preload/resources/fetch-destination-worker.js

```diff
@@ -0,0 +1,11 @@
+self.addEventListener('fetch', function(event) {
+    if (event.request.url.indexOf('dummy.xml') != -1) {
+        var response = new Response();
+        if (event.request.destination == "")
+            response.status = 200;
+        else
+            response.status = 500;
```

Comment:
Hmm, I guess we need to tighten the definition of "load was successful" there. I understood it as including the status code. Is there another way that a SW can signal the page that a load was unsuccessful?